Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -61,14 +66,109 @@ - (BOOL)installWithOptions:(nonnull SentryOptions *)options
 
     [SentryGlobalEventProcessor.shared
         addEventProcessor:^SentryEvent *_Nullable(SentryEvent *_Nonnull event) {
-            [self.sessionReplay captureReplayForEvent:event];
-
+            if (event.isCrashEvent) {
+                [self resumePreviousSessionReplay:event];
+            } else {
+                [self.sessionReplay captureReplayForEvent:event];
+            }
             return event;
         }];
 
     return YES;
 }
 
+- (void)resumePreviousSessionReplay:(SentryEvent *)event
+{
+    NSURL *dir = [self replayDirectory];
+    NSData *lastReplay =
+        [NSData dataWithContentsOfURL:[dir URLByAppendingPathComponent:@"lastreplay"]];
+    if (lastReplay == nil) {
+        return;
+    }
+
+    NSDictionary<NSString *, id> *jsonObject = [NSJSONSerialization JSONObjectWithData:lastReplay
+                                                                               options:0
+                                                                                 error:nil];
+    SentryId *replayId = jsonObject[@"replayId"]
+        ? [[SentryId alloc] initWithUUIDString:jsonObject[@"replayId"]]
+        : [[SentryId alloc] init];
+    NSURL *lastReplayURL = [dir URLByAppendingPathComponent:jsonObject[@"path"]];
+
+    SentryCrashReplay crashInfo = { 0 };
+    bool hasCrashInfo = sentrySessionReplaySync_readInfo(&crashInfo,
+        [[lastReplayURL URLByAppendingPathComponent:@"crashInfo"].path
+            cStringUsingEncoding:NSUTF8StringEncoding]);
+
+    SentryReplayType type = hasCrashInfo ? SentryReplayTypeSession : SentryReplayTypeBuffer;
+    NSTimeInterval duration
+        = hasCrashInfo ? _replayOptions.sessionSegmentDuration : _replayOptions.errorReplayDuration;
+    int segmentId = hasCrashInfo ? crashInfo.segmentId + 1 : 0;
+
+    _resumeReplayMaker = [[SentryOnDemandReplay alloc] initWithContentFrom:lastReplayURL.path];
+    _resumeReplayMaker.bitRate = _replayOptions.replayBitRate;
+    _resumeReplayMaker.videoScale = _replayOptions.sizeScale;
+
+    NSDate *beginning = hasCrashInfo
+        ? [NSDate dateWithTimeIntervalSinceReferenceDate:crashInfo.lastSegmentEnd]
```

Comment:
The existence of `crasInfo.lastSegmentEnd` means the app crashed while session replay was already in session mode and running. 
However, session replay can also run in buffer mode, which means it is not transmitting replays but saving screenshots to use in case of an error. If the app crashes, we need to send the replay of the crash.